pennylane/transforms/hamiltonian_expand.py

```diff
@@ -341,6 +341,8 @@ def sum_expand(tape: QuantumTape, group: bool = True) -> (Sequence[QuantumTape],
     idxs_coeffs_dict = {}  # {m_hash: [(location_idx, coeff)]}
     for idx, m in enumerate(tape.measurements):
         obs = m.obs
+        if isinstance(obs, Prod) and isinstance(m, ExpectationMP):
+            obs = obs.simplify()
```

Comment:
What breaks without this? Can we add the corresponding test that would fail without this to the test suite?

Reply:
Was also wondering this 🤔

Reply:
I would only keep this if there is a very good reason for it, as simplification can be rather expensive